dal/TipsDao.java

```diff
@@ -0,0 +1,260 @@
+package aireats.dal;
+
+import aireats.model.*;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Data access object (DAO) class to interact with the underlying Persons table
+ * in MySQL instance. This is used to store {@link Tips} into MySQL instance and
+ * retrieve
+ * {@link Tips} from MySQL instance.
+ */
+public class TipsDao {
+    protected ConnectionManager connectionManager;
+
+    // Single pattern: instantiation is limited to one object.
+    private static TipsDao instance = null;
+
+    protected TipsDao() {
+        connectionManager = new ConnectionManager();
+    }
+
+    public static TipsDao getInstance() {
+        if (instance == null) {
+            instance = new TipsDao();
+        }
+        return instance;
+    }
+
+    /**
+     * Save the Tips instance by storing it in MySQL instance.
+     * This runs a INSERT statement.
+     */
+    public Tips create(Tips tips) throws SQLException {
+        String insertTip = "INSERT INTO Tips(UserId, RestaurantId, Compliment_count, Date, Context) VALUES(?,?,?,?,?);";
+        Connection connection = null;
+        PreparedStatement insertStmt = null;
+        ResultSet resultKey = null;
+        try {
+            connection = connectionManager.getConnection();
+            insertStmt = connection.prepareStatement(insertTip);
+            insertStmt.setString(1, tips.getUserId());
+            insertStmt.setString(2, tips.getRestaurantId());
+            insertStmt.setInt(3, tips.getCompliment_count());
+            insertStmt.setTimestamp(4, new Timestamp(tips.getDate().getTime()));
```

Comment:
Do we want to change the data types of "date" column in create table statements to be of "Date" type, currently it's of "Datetime" type. If so, we can change this line into ```insertStmt.setDate(4, tips.getDate()); ```

Reply:
It would be better to keep datatype unchanged for the "Tips" table. I'll modify my code.